app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala

```diff
@@ -37,25 +38,36 @@ class GetMessagesController @Inject()(
                                      )(implicit ec: ExecutionContext)
   extends BackendController(cc) {
 
-  def getMessagesForMovement(lrn: String): Action[AnyContent] = {
+  def getMessagesForMovement(lrn: String, updatedSince: Option[String]): Action[AnyContent] = {
     // todo: how we handle error here if for example MongoDb throws?
     authAction.async(parse.default) {
       implicit request: EnrolmentRequest[AnyContent] => {
 
         movementService.getMatchingERN(lrn, request.erns.toList).flatMap {
           case None => Future.successful(BadRequest(Json.toJson(ErrorResponse(LocalDateTime.now(), "Invalid LRN supplied for ERN", ""))))
           case Some(ern) => workItemService.addWorkItemForErn(ern, fastMode = false)
-            getMessagesAsJson(lrn, ern)
+            getMessagesAsJson(lrn, ern, updatedSince)
         }
       }
     }
   }
 
-  private def getMessagesAsJson(lrn: String, ern: String): Future[Result] = {
-    movementService.getMovementByLRNAndERNIn(lrn, List(ern)).map {
-      case Some(mv) => Ok(Json.toJson(mv.messages))
+  private def getMessagesAsJson(lrn: String, ern: String, updatedSince: Option[String]): Future[Result] = {
+    val updatedSinceTime = updatedSince.map(Instant.parse(_))
```

Comment:
What happens if the Instant isn't parsed properly, I can't see the error handling code, is this converted to an Internal Server Error or is the Exception shown to the user? If it fails to parse we should probably return a BadRequest with our standard Error Response?

Reply:
I'll do this instead yeah. Might as well do it now.